### DISCUSSION TITLE
Single option on IOS

### DISCUSSION THREAD

**Discussion by BlagojeV93** [WalletConnect/walletconnect-monorepo#1441] [Developer Support]
Hello all!

I want to thank all the contributors for this amazing library, I've been using it for couple of months, but still been unable to find a solution (if there is) for this.

Inside my react-native app, I use standard connector object (with useWalletConnect hook) to connect to a MetaMask wallet. Because that's the only option I need so far, I was wandering is there an option to skip the "wallet connect modal" part, and just to use Metamask connect option and send the users to their MetaMask wallet in order to connect?
Inside this modal there are several options (like Pillar or imToken) that appear to not be working (clicking on those isn't redirecting user anywhere on my IOS device). That's the main reason I would like to have something like a universal MetaMask method that's initializing connection only to MetaMask wallets.

Any help would be appreciated. Thanks!

**Comment by chadyj** [comment]
hey! :wave: This is pretty straight forward. You can build your own "modal" with 1 just one wallet.

You can get the deep/universal links from our Explorer API for Metamask here https://docs.walletconnect.com/2.0/introduction/cloud-explorer#cloud-explorer-api

Then check this page for details on how to build the URI https://docs.walletconnect.com/mobile-linking#for-ios
